music21/environment.py

```diff
@@ -613,17 +611,43 @@ def getTempFile(self, suffix='', returnPathlib=True) -> Union[str, pathlib.Path]
         >>> import pathlib
         >>> isinstance(e.getTempFile(), pathlib.Path)
         True
+
+        >>> oldScratchDir = e['directoryScratch']
+        >>> e['directoryScratch'] = None
+        >>> import getpass, os, stat
+        >>> os.chmod(e.getDefaultRootTempDir(), stat.S_IREAD)  # wipe out write, exec permissions
+        >>> 'music21-' + getpass.getuser() in e.getTempFile(returnPathlib=False)
+        True
+        >>> os.chmod(e.getDefaultRootTempDir(), stat.S_IREAD|stat.S_IWRITE|stat.S_IEXEC)  # restore
+        >>> e['directoryScratch'] = oldScratchDir  # restore
```

Comment:
I think that this is a little too complex for the doctest for this -- can we move to a unittest and put the restoration of directoryScratch into a "finally" clause of a try: so that if previous tests fail for some reason, we still restore the old test?